kin-ecosystem-sdk/src/main/java/com/kin/ecosystem/balance/view/BalanceView.java

```diff
@@ -1,120 +1,260 @@
-    }
-
-    @Override
-    protected void onAttachedToWindow() {
-        super.onAttachedToWindow();
-        this.balancePresenter.onAttach(this);
-    }
-
-    @Override
-    protected void onDetachedFromWindow() {
-        super.onDetachedFromWindow();
-        this.balancePresenter.onDetach();
-    }
-
-    @Override
-    public void updateBalance(final String balance) {
-        if (this.balanceText != null) {
-            post(new Runnable() {
-                @Override
-                public void run() {
-                    balanceText.setText(balance);
-                }
-            });
-        }
-    }
-
-    @Override
-    public void updateSubTitle(final String sub_title) {
-        if (this.subTitle != null) {
-            post(new Runnable() {
-                @Override
-                public void run() {
-                    subTitle.setText(sub_title);
-                }
-            });
-        }
-    }
+	private static final int ANIM_DURATION = 200;
+	private static final float FLOAT_1 = 1f;
+	private static final float FLOAT_0 = 0f;
+	private static final float FLOAT_08 = 0.8f;
+	private static int blueColor;
+	private static int orangeColor;
+	private static int redColor;
+	private static int subTitleColor;
+
+	private static float balanceTextSize;
+	private static float subTitleTextSize;
+
+	private IBalancePresenter balancePresenter;
+
+	private TextSwitcher subTitleTextView;
+	private TextSwitcher balanceText;
+	private ImageView arrow;
+
+	private float arrowFromX = 0;
+	private float arrowToX = 0;
+
+	public BalanceView(Context context) {
+		super(context);
+		init(context, null);
+	}
+
+	public BalanceView(Context context, AttributeSet attrs) {
+		super(context, attrs);
+		init(context, attrs);
+	}
+
+	public BalanceView(Context context, AttributeSet attrs, int defStyleAttr) {
+		super(context, attrs, defStyleAttr);
+		init(context, attrs);
+	}
+
+	private void init(Context context, @Nullable AttributeSet attrs) {
+		inflate(context, R.layout.kinecosystem_balance_view, this);
+		setLayoutParams(new ConstraintLayout.LayoutParams(ViewGroup.LayoutParams.MATCH_PARENT,
+			ViewGroup.LayoutParams.WRAP_CONTENT));
+		int topPadding = getResources().getDimensionPixelOffset(R.dimen.kinecosystem_main_medium_margin);
+		setPadding(0, topPadding, 0, 0);
+		initColors();
+		initSizes();
+
+		TypedArray styledAttributes = context.getTheme()
+			.obtainStyledAttributes(attrs, R.styleable.KinEcosystemBalanceView, 0, 0);
+		String subtitle;
+		try {
+			subtitle = styledAttributes.getString(R.styleable.KinEcosystemBalanceView_subTitle);
+
+		} finally {
+			styledAttributes.recycle();
+		}
+
+		this.subTitleTextView = findViewById(R.id.sub_title);
+		this.balanceText = findViewById(R.id.balance_text);
+		this.arrow = findViewById(R.id.arrow);
+		addBalanceViewFactory();
+		addSubTitleViewFactory();
+		setOnClickListener(new OnClickListener() {
+			@Override
+			public void onClick(View v) {
+				balancePresenter.balanceClicked();
```

Comment:
you can call both balancePresenter.balanceClicked(); and ecosystemPresenter.balanceItemClicked() from here.
No need for the back and forth

Reply:
one view can't handle too presenters, also we don't have the `ecosystemPresenter` on this view.

Reply:
ok